File: assignment3/leg.py
class Leg:
    DEFAULT_LENGTH : float = 20.0 # inches
    MAX_WEIGHT : float = 75 # pounds per leg

    def __init__(self, length : float = DEFAULT_LENGTH):
        self._length = length
        self._broken = False
    
    @property 
    def length(self) -> float:
        return self._length

    @property
    def broken(self) -> bool:
        return self._broken
    
    @broken.setter
    def broken(self, value : bool) -> None:
        self._broken = value

File: assignment3/c++/src/leg.cpp
#include <stdexcept>
#include "leg.h"

const float Leg::DEFAULT_LENGTH = 20.0; //inches
const float Leg::MAX_WEIGHT = 75.0; //pounds

Leg::Leg(float length) {
    Leg::_length = length;
    Leg::_broken = false;
    if(length < 0)
        throw std::invalid_argument("Length must be > 0");
}

float Leg::getLength() const { return _length; }
bool Leg::isBroken() const { return _broken; } 

void Leg::setBroken(bool broken) {
    Leg::_broken = broken;
}
